lms/djangoapps/mobile_api/mobile_platform.py

```diff
@@ -0,0 +1,59 @@
+"""
+Platform related Operations for Mobile APP
+"""
+import abc
+import re
+
+
+class MobilePlatform:
+    """
+    MobilePlatform class creates an instance of platform based on user agent and supports platform
+    related operations
+    """
+    __metaclass__ = abc.ABCMeta
+    version = None
+
+    def __init__(self, version):
+        self.version = version
+
+    @classmethod
+    def get_user_app_platform(cls, user_agent, user_agent_regex):
+        """ Returns platform instance if user_agent matches with USER_AGENT_REGEX """
+        match = re.search(user_agent_regex, user_agent)
```

Comment:
regex search will only return the matched regex; while we want to return the instance of platform class if the regex gets matched